### ISSUE TITLE
Causal inference in R

### ISSUE THREAD

**Issue by emilyriederer** [chirunconf/chirunconf19#5]
Opening a general issue on this idea as an area for discussion. I believe numerous participants expressed interest or had experience in this field. Clearly, projects here could take a lot of different shapes, but let's use this thread to brainstorm further!

**Comment by benlistyg** [comment]
One idea I have for this is to contribute to the `DeclareDesign` [package](https://declaredesign.org/) from Blair et al. They could use a vignette for a Two-Arm Covariate Design (see [here](https://declaredesign.org/library/)).

**Comment by emilyriederer** [comment]
Neat idea! They have such rich documentation in most places; anything to make it even better would be a real win

**Comment by weiyangtham** [comment]
A general thought I've had is that it would be nice to shepherd some causal inference packages into the tidyverse, mainly by making the output extract-able from broom (alternatively contributing a method to DeclareDesign). A few off the top of my head: 

- [`rdrobust`](https://sites.google.com/site/rdpackages/rdrobust) for regression discontinuity
- [`cem`](https://cran.r-project.org/web/packages/cem/index.html) for Coarsened Exact Matching
- [`gsynth`](http://yiqingxu.org/software/gsynth/gsynth_examples.html) and [`synth`](https://cran.r-project.org/web/packages/Synth/Synth.pdf) for synthetic control

**Comment by alexpghayes** [comment]
As someone just getting more into causal inference I'd also love to see an overview of packages for causal inference in R. I feel like many methods are siloed in packages maintained by individual research groups that don't talk to each other, so it'd be good to know:

- what methods are implemented
- where do they live
- which could possibly be unified / made consistent with each other

I'm especially curious about the approach in [DoWhy](https://github.com/Microsoft/dowhy), where identification and estimation are separate steps in a modeling workflow.

**Comment by emilyriederer** [comment]
@weiyangtham This is a great idea! I know the [`cobalt`](https://cran.r-project.org/web/packages/cobalt/index.html) package tries to do something sort of similar to `broom` by just providing standardized outputs from many different packages (mostly in the more "classical" weighting / matching space). 

It's not in the "tidy" paradigm, so we could either use it for inspiration for making new additions to broom or also consider making a PR to `cobalt` (authors willing) to include techniques beyond weighting / matching?

**Comment by emilyriederer** [comment]
@alexpghayes Wow - I had ne ver heard of `DoWhy` before at all! Thanks for sharing! This looks really interesting 😎

**Comment by weiyangtham** [comment]
@alexpghayes Yes! Are you thinking of something like a causal inference CRAN task view? DoWhy looks super cool and in the long-term a unifying framework would be ideal

**Comment by weiyangtham** [comment]
@emilyriederer I hadn't known of `cobalt` but what you said definitely seems like the right direction! Will play around with it a bit
